File: LeetCode/LeetCode Practice Problems/205- Isomorphic Strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        map<char,char> m1,m2;
        
        int n = s.length();
        
        for(int i=0;i<n;i++) {
            if(m1.find(s[i])!=m1.end() && m1[s[i]] != t[i]) return false;
            if(m2.find(t[i])!=m2.end() && m2[t[i]] != s[i]) return false;
            m1[s[i]] = t[i];
            m2[t[i]] = s[i];
        }
        
        return true;
        
    }
};

File: LeetCode/LeetCode Practice Problems/205-Isomorphic Strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        Map<Character,Character> m1 = new HashMap();
        Map<Character,Character> m2 = new HashMap();
        for(int i=0;i<s.length();i++) {
            char ch1 = s.charAt(i);
            char ch2 = t.charAt(i);
            if((m2.containsKey(ch2) && m2.get(ch2) != ch1) || (m1.containsKey(ch1) && m1.get(ch1) != ch2)) {
                return false;
            }
            m1.put(ch1 , ch2);
            m2.put(ch2 , ch1);
        }
        return true;
    }
}
